sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Projection.scala

```diff
@@ -74,6 +78,43 @@ case class InterpretedMutableProjection(expressions: Seq[Expression]) extends Mu
 }
 
 /**
+ * A projection that could turn any InternalRow into UnsafeRow
+ */
+case class ToUnsafeProjection(fields: Seq[DataType]) extends Projection {
```

Comment:
Do we need this? If we are in interpretation mode, we might as well not do anything Unsafe.